ddtrace/tracer/post_processor.go

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package tracer
+
+import (
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+var _ ddtrace.ReadWriteSpan = (*readWriteSpan)(nil)
+
+// readWriteSpan wraps span and implements the ddtrace.ReadWriteSpan interface.
+type readWriteSpan struct {
+	*span
+}
+
+// Tag returns the tag value held by the given key.
+func (s readWriteSpan) Tag(key string) interface{} {
+	s.Lock()
+	defer s.Unlock()
+
+	switch key {
+	// String.
+	case ext.SpanName:
+		return s.Name
+	case ext.ServiceName:
+		return s.Service
+	case ext.ResourceName:
+		return s.Resource
+	case ext.SpanType:
+		return s.Type
+	// Bool.
+	case ext.AnalyticsEvent:
+		return s.Metrics[ext.EventSampleRate] == 1.0
+	case ext.ManualDrop:
+		return s.Metrics[keySamplingPriority] == -1
+	case ext.ManualKeep:
+		return s.Metrics[keySamplingPriority] == 2
+	// Metrics.
+	case ext.SamplingPriority, keySamplingPriority:
+		if val, ok := s.Metrics[keySamplingPriority]; ok {
+			return val
+		}
+		return nil
+	}
+	if val, ok := s.Meta[key]; ok {
+		return val
+	}
+	if val, ok := s.Metrics[key]; ok {
+		return val
+	}
+	return nil
+}
+
+// IsError reports wether s is an error.
+func (s readWriteSpan) IsError() bool {
+	s.Lock()
+	defer s.Unlock()
+
+	return s.Error == 1
+}
+
+// SetOperationName is a no-op.
```

Comment:
It doesn't look like a no-op?

Reply:
This will match the case [here](https://github.com/DataDog/dd-trace-go/blob/a96015d3909a2c8f319f63ec701c92c548f15e96/ddtrace/tracer/post_processor.go#L80):
```
case ext.SpanName, ext.SpanType, ext.ResourceName, ext.ServiceName, ext.HTTPCode, ext.Environment, keyMeasured, keyTopLevel, ext.AnalyticsEvent, ext.EventSampleRate:
	log.Debug("Setting the tag %v in the processor is not allowed", key)
	return
```
which will log that this is not allowed and return without changing the operation name. Agreed that this is not fully a no-op though, and the description needs to be clarified 🙂 . What do you think of this wording:
```
// SetOperationName is not allowed in the processor and will not modify the operation name.
```
?

Reply:
Maybe this instead then?

```
// SetOperationName is not allowed in the processor and will not modify the operation name.
func (s readWriteSpan) SetOperationName(operationName string) {
	s.SetTag(ext.SpanName, nil) // this will log a debug line and be a no-op
}
```

No reason to pass in the operationName to SetTag if we aren't going to use it.

Reply:
I would not advise doing that. It is making assumptions about other code which may or may not hold true in the future. I wouldn't want us to end up having the operation name removed by accident as that would be a pretty tragic occurrence. Maybe it's better we do nothing here and just log the debug message?

Reply:
Yeah just logging the debug message is certainly the safest bet. The code duplication would be very minor. (Though ideally we would also have tests that would catch a regression like that either way)

Reply:
Sounds good, will change this to only log the debug message!